plugins/flytekit-aws-sagemaker/flytekitplugins/awssagemaker_inference/agent.py

```diff
@@ -0,0 +1,116 @@
+import json
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+from flytekit import FlyteContextManager
+from flytekit.core.type_engine import TypeEngine
+from flytekit.extend.backend.base_agent import (
+    AgentRegistry,
+    AsyncAgentBase,
+    Resource,
+    ResourceMeta,
+)
+from flytekit.extend.backend.utils import convert_to_flyte_phase, get_agent_secret
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+
+from .boto3_mixin import Boto3AgentMixin
+
+states = {
+    "Creating": "Running",
+    "InService": "Success",
+    "Failed": "Failed",
+}
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+
+        return json.JSONEncoder.default(self, o)
+
+
+@dataclass
+class SageMakerEndpointMetadata(ResourceMeta):
+    endpoint_name: str
+    region: str
+
+
+class SageMakerEndpointAgent(Boto3AgentMixin, AsyncAgentBase):
+    """This agent creates an endpoint."""
+
+    name = "SageMaker Endpoint Agent"
+
+    def __init__(self):
+        super().__init__(
+            service="sagemaker",
+            task_type_name="sagemaker-endpoint",
+            metadata_type=SageMakerEndpointMetadata,
+        )
+
+    async def create(
+        self, task_template: TaskTemplate, inputs: Optional[LiteralMap] = None, **kwargs
+    ) -> SageMakerEndpointMetadata:
+        custom = task_template.custom
+        config = custom.get("config")
+        region = custom.get("region")
+
+        await self._call(
+            method="create_endpoint",
+            config=config,
+            inputs=inputs,
+            region=region,
+            aws_access_key_id=get_agent_secret(secret_key="aws-access-key"),
+            aws_secret_access_key=get_agent_secret(secret_key="aws-secret-access-key"),
+            aws_session_token=get_agent_secret(secret_key="aws-session-token"),
+        )
```

Comment:
Should we change the secret to  `flyte_xxxx_xxxx_xxxx` ?
For example, `flyte_openai_access_token `?
cc @pingsutw